mscore/seq.cpp

```diff
@@ -1405,10 +1415,13 @@ void Seq::putEvent(const NPlayEvent& event, unsigned framePos)
             qDebug("bad channel value %d >= %d", channel, cs->midiMapping()->size());
             return;
             }
+
+      // dispatch midi messages to MIDI driver first before rendering with MuseScore's synth, in order to avoid additional unpredictable latency associated with MuseScore's synth
+      if (_driver != 0 && (preferences.useJackMidi || preferences.useAlsaAudio || preferences.usePortaudioAudio))
+            _driver->putEvent(event, framePos);
```

Comment:
hmm...apparently takeLast() doesn't delete...but I'm not totally sure https://forum.qt.io/topic/34233/qlist-takelast-and-free-memory/3